export-tiles/src/main/java/com/esri/arcgisruntime/sample/exporttiles/MainActivity.java

```diff
@@ -189,12 +190,17 @@ private void initiateDownload() {
         .createDefaultExportTileCacheParametersAsync(viewToExtent(), minScale, maxScale);
     parametersFuture.addDoneListener(new Runnable() {
       @Override public void run() {
+        // create directory for file
+        File file = new File(Environment.getExternalStorageDirectory(),
+            getString(R.string.config_data_sdcard_offline_dir));
+        mDirExists = file.exists();
+        if (mDirExists) {
+          Log.i(TAG, "Directory exists");
+        } else {
+          file.mkdirs();
```

Comment:
Still not using the result here, refactor to `mDirExists = file.mkdirs();`.  Since the file i/o operation can fail best to fully check the results: 

```java
mDirExists = file.mkdirs();
if(mDirExists){
  Log.i(TAG, "Created directory");
}else{
  Log.d(TAG, "error!!");
}
```

Reply:
Ahh ok I see now that it not only creates the file, but also returns a boolean about creating the file! Pretty cool...

But I'm not sure I see why this should be an error on !mDirExists. If the directory already exists, thats fine. It may well exist for other sample data. I'm just trying to insure that the directory does exist by the time the app wants to write something to it.

If I implement the code as above, I get "error!!" every time I click Export Tiles.